src/Storage/Shard/Redis/Connection.php

```diff
@@ -151,14 +151,47 @@ public function connect() : \Maleficarum\Storage\Shard\ShardInterface {
     /**
      * @see https://github.com/phpredis/phpredis#scan
      */
-    public function scan(&$cursor) {
+    public function scan(&$cursor, $pattern = null, $count = 0) {
```

Comment:
Default value for the pattern parameter seems incorrect based on both the documentation for phpredis and your own code for hScan, zScan and so on.

Reply:
Improved in 46bef5a5b9ddc84c86fddc1f751a9490fc2dc033